lib/octokit/client/stats.rb

```diff
@@ -75,11 +75,32 @@ def punch_card_stats(repo, options = {})
       #
       # @param repo [Integer, String, Hash, Repository] A GitHub repository
       # @param metric [String] The metrics you are looking for
-      # @return [Array<Sawyer::Resource>] Magical unicorn stats
+      # @return [Array<Sawyer::Resource> or nil] Stats in metric-specific format, or nil if not yet calculated.
+      # @see https://developer.github.com/v3/repos/statistics/
       def get_stats(repo, metric, options = {})
-        data = get("#{Repository.path repo}/stats/#{metric}", options)
+        path = "#{Repository.path repo}/stats/#{metric}"
+        if retry_timeout = options.delete(:retry_timeout)
+          get_stats_with_retry(retry_timeout, path, options)
+        else
+          get(path, options)
+        end.tap do
+          return nil if last_response.status == 202
```

Comment:
With retry or not, we can still end up with `202`. `#get` returns an empty string in such case and we need to convert this empty string to `nil`. At least this is what old version did, see [this line](https://github.com/octokit/octokit.rb/pull/687#discussion-diff-48182966L82).